app/data_model/answer_store.py

```diff
@@ -245,7 +249,24 @@ def upgrade_1_to_2_add_group_instance_id(answer_store, schema):
         answer['group_instance_id'] = get_group_instance_id(schema, answer_store, location, answer['answer_instance'])
 
 
-UPGRADE_TRANSFORMS = (
-    upgrade_0_to_1_update_date_formats,
-    upgrade_1_to_2_add_group_instance_id,
-)
+def upgrade_3_to_4_remove_empty_answers(answer_store, schema):  # pylint: disable=unused-argument
+    """ Previously, we stored [] and '' as answer values for unanswered, but completed answers.
+
+    This was changed to avoid storing unanswered values entirely, so this upgrade will
+    remove any answers in the store that have 'empty' answer values
+    """
+    answer_store_copy = answer_store.copy()
+
+    for answer in answer_store:
+        if answer['value'] in ('', [], None):
+            answer_store_copy.remove_answer(answer)
+
+    answer_store = answer_store_copy
+
+
+# Dictionary specifying upgrade methods. Key should be the version to upgrade from. It is assumed that each transform only upgrades the version by 1.
+UPGRADE_TRANSFORMS = {
+    0: upgrade_0_to_1_update_date_formats,
+    1: upgrade_1_to_2_add_group_instance_id,
+    3: upgrade_3_to_4_remove_empty_answers,
```

Comment:
I think it would be clearer if the key was the version the upgrade is for, so would be 1,2,4 rather than 0,1,3. 

We should also drop the from version numbers from the method names as they don't make sense now version numbers can be skipped. For example `upgrade_3_to_4_remove_empty_answers` also upgrades version 2 answer stores.

Reply:
Fixed.